src/compiler/checker.ts

```diff
@@ -4109,7 +4109,7 @@ namespace ts {
         function getExportSymbolOfValueSymbolIfExported(symbol: Symbol): Symbol;
         function getExportSymbolOfValueSymbolIfExported(symbol: Symbol | undefined): Symbol | undefined;
         function getExportSymbolOfValueSymbolIfExported(symbol: Symbol | undefined): Symbol | undefined {
-            return getMergedSymbol(symbol && (symbol.flags & SymbolFlags.ExportValue) !== 0 ? symbol.exportSymbol : symbol);
+            return getMergedSymbol(symbol && (symbol.flags & SymbolFlags.ExportValue) !== 0 && symbol.exportSymbol || symbol);

```

Comment:
The symbol passed here that had the `ExportValue` flag but no `exportSymbol` was the `export=` symbol for `constants.js` in the test case.

Reply:
The `export=` symbol is the result of a symbol merge, afaik - is it as simple as `cloneSymbol` not copying this field when it should?

Reply:
This `export=` symbol is not a merged symbol, no.